components/raftstore/src/store/peer.rs

```diff
@@ -2233,18 +2235,30 @@ impl Peer {
         req: &mut RaftCmdRequest,
     ) -> Result<()> {
         let last_index = self.raft_group.raft.raft_log.last_index();
-        let min_progress = self.get_min_progress()?;
-        let min_index = min_progress + 1;
-        if min_progress == 0 || last_index - min_progress > ctx.cfg.merge_max_log_gap {
+        let (min_matched, min_committed) = self.get_min_progress()?;
+        assert!(min_matched >= min_committed);
```

Comment:
`min_matched` can be reset to 0, however `min_committed` won't be. The assert should be moved to L2251.